### PR TITLE
Use Mockito instead of EasyMock

### PR DESCRIPTION
**Pull request by matozoid**
I saw you poor symbol solver people had to use horrible, horrible EasyMock, so I gave you the glory of Mockito instead.

### CODE REVIEW

**Comment by matozoid** [comment]
Intellij was completely confused with all the test java files by the way, thinking they were real source code.

**Comment by ftomassetti** [comment]
Regarding IntelliJ: I think you can mark the problematic subdirectories as "excluded" and it should stop presenting them. Note that IntelliJ is doing the wrong thing because it is considerign for compilation sources that are under in the resources directory

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/ContextTest.java [hunk 1]
```diff
@@ -162,22 +159,20 @@ public void resolveReferenceToClassesInTheSamePackage() throws ParseException {
...
+        ResolvedClassDeclaration compilationUnitDecl = mock(ResolvedClassDeclaration.class);
+        when(compilationUnitDecl.getName()).thenReturn("CompilationUnit");
+        when(compilationUnitDecl.getQualifiedName()).thenReturn("my.packagez.CompilationUnit");
+        TypeSolver typeSolver = mock(TypeSolver.class);
+        when(typeSolver.getRoot()).thenReturn(typeSolver);
+        when(typeSolver.solveType("java.lang.Object")).thenReturn(new ReflectionClassDeclaration(Object.class, typeSolver));
+        when(typeSolver.tryToSolveType("my.packagez.CompilationUnit")).thenReturn(SymbolReference.solved(compilationUnitDecl));
         SymbolSolver symbolSolver = new SymbolSolver(typeSolver);
-        replay(typeSolver, compilationUnitDecl);
+
```
Comment (ftomassetti) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/ContextTest.java:170]: No need to specify the replay here?

**Review by ftomassetti — APPROVED** [review]
I am convinced!

**Pull request merged**